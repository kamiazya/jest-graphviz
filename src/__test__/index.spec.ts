import '../index';
describe('public api', () => {
  const runOnWindows = process.platform === 'win32';

  const dot = 'digraph g { a -> b; }';

  // Skip test on windows
  // NOTE: Not support output json format on Graphviz windows statable now.
  (runOnWindows ? test.skip : test)('toMatchDotJsonSnapshot', () => {
    expect(dot).toMatchDotJsonSnapshot();
  });

  test('toBeValidDotAndMatchSnapshot', () => {
    expect(dot).toBeValidDotAndMatchSnapshot();
  });
  test('toBeValidDot', () => {
    expect(dot).toBeValidDot();
  });
});
