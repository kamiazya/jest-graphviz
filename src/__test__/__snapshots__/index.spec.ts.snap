// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`public api toBeValidDotAndMatchSnapshot 1`] = `digraph g { a -> b; }`;

exports[`public api toMatchDotJsonSnapshot 1`] = `
Object {
  "_subgraph_cnt": 0,
  "directed": true,
  "edges": Array [
    Object {
      "_gvid": 0,
      "head": 1,
      "tail": 0,
    },
  ],
  "name": "g",
  "objects": Array [
    Object {
      "_gvid": 0,
      "label": "\\\\N",
      "name": "a",
    },
    Object {
      "_gvid": 1,
      "label": "\\\\N",
      "name": "b",
    },
  ],
  "strict": false,
}
`;
